ili9341.cpp

```diff
@@ -261,13 +261,22 @@ ili9341_init(void)
 	delay(5);
   }
 }
-
+#if 0
 static inline uint32_t __REV16(uint32_t rev)
 {
 	uint32_t a = __builtin_bswap16(rev >> 16);
 	uint32_t b = __builtin_bswap16(rev & 0xFFFF);
 	return (a << 16) | (b);
 }
+#else
+// Reverses the byte order within each halfword of a word. For example, 0x12345678 becomes 0x34127856.
+static inline uint32_t __REV16(uint32_t value)
+{
+  uint32_t result;
+  __asm volatile("rev16 %0, %1" : "=r" (result) : "r" (value));

```

Comment:
I suggest moving inline assembly stuff to its own file, maybe a utility.hpp that contains these inline functions. In the future I think there are going to be risc-v based instruments.

Reply:
If multiple cpu architectures need to be supported then we should keep this. Utility.hpp seems fine.

Just checking: there is no direct built in for this?